File: OpenGL/Hull Invaders/SpaceInvaders/SpaceInvaders/Timer.cs
using System;
using System.Collections.Generic;
using System.Text;
using Tao.OpenGl;
using Tao.Glut;
using System.Collections;
using System.Windows.Forms;

namespace SpaceInvaders
{
    public interface ITimer
    {
        /// <summary>
        /// Returns the amount of time in seconds that has elapsed since the program started
        /// </summary>
        /// <returns>A float</returns>
        float GetRunTime();

        /// <summary>
        /// Returns the amount of time in seconds that has elapsed since the last time that
        /// this function was called
        /// </summary>
        /// <returns>A float</returns>
        float Tick();
    }

    public class Timer : ITimer
    {
        private float ticksSinceStart;
        private float oldTick;

        public Timer()
        {
            ticksSinceStart = oldTick = System.Environment.TickCount;
        }

        public float GetRunTime()
        {
            ticksSinceStart = (float)System.Environment.TickCount;
            return (float)ticksSinceStart;
        }

        public float Tick()
        {
            ticksSinceStart = (float)(System.Environment.TickCount);
            float dt = ((float)(ticksSinceStart-oldTick)/(float)System.Environment.TickCount);
            oldTick = ticksSinceStart;

            return dt;

        }
    }
}

File: C++/Sony PSP/Particle Engine/Prototype_03/Prototype_01/Timer.cpp
#include "Timer.h"
Timer::Timer(void)
{
	_ticksSinceStart = clock();
	_oldTick = clock();
	_ticksSinceCreation = clock();
	_timeofCreation = clock();
}
Timer::~Timer(void)
{

}
float Timer::GetRunTime() // returns the amount of time in seconds that has elapsed since the program
// started.
{
	_ticksSinceStart = clock();
	return ((float)_ticksSinceStart/(float)CLOCKS_PER_SEC);
}
float Timer::Tick() // returns the amount of time in seconds that has elapsed since the last time that
// this same function was called.
{
	_ticksSinceStart = clock();
	float dt = ((float)(_ticksSinceStart-_oldTick)/(float)CLOCKS_PER_SEC);
	_oldTick = _ticksSinceStart;
	return dt;
}

float Timer::GetCreationTime()
{
   _ticksSinceCreation = clock();
   float dt = ((float)(_ticksSinceCreation - _timeofCreation) / (float)CLOCKS_PER_SEC);
   return dt;
}

void Timer::ResetTimer()
{
	_ticksSinceStart = clock();
	_oldTick = clock();
	_ticksSinceCreation = clock();
	_timeofCreation = clock();
}